src/shared/identityUtils.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { CryptoUtils } from '@salesforce/lwc-dev-mobile-core';
+import { Config, ConfigAggregator } from '@salesforce/core';
+import configMeta, { ConfigVars } from './../configMeta.js';
+
+export class IdentityUtils {
+  public static async updateConfigWithIdentityToken(): Promise<void> {
+    const token = await this.getIdentityToken();
+    if (!token) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
+      const generatedIdentityToken = CryptoUtils.generateIdentityToken();
+      await this.writeIdentityToken(generatedIdentityToken);
+    }
+  }
+
+  public static async getIdentityToken(): Promise<string | undefined> {
+    const config = await ConfigAggregator.create({ customConfigMeta: configMeta });
+    // Need to reload to make sure the values read are decrypted
+    await config.reload();
```

Comment:
@maliroteh-sf @khawkins Need to use `ConfigAggregator` instead of simple `Config` to read the token. The reason is because `Config`'s `get(<token name>, <decrypt flag>)` fails to decrypt the value read from the config file. And the only way is to use `ConfigAggregator` and then to `reload()`.

Reply:
Hmm... that's too bad. Probably need to follow up in [platform-cli-collaboration](https://salesforce.enterprise.slack.com/archives/C06V045BZD0) to see if this is a known limitation, a bug, or are we missing something. But that can be done in the future too.

Reply:
I pinged one of the engineers earlier and was offered a potential solution but have got to try it yet. Not sure if it will work. I'll make another PR if it does and see if I could address your concern about being able to pass the config.